src/Fixer/PhpTag/ShortEchoTagFixer.php

```diff
@@ -0,0 +1,242 @@
+    const LONG_FUNCTION_PRINT = 'print';
+
+    private $supportedFormatOptions = [
+        self::FORMAT_LONG,
+        self::FORMAT_SHORT,
+    ];
+
+    private $supportedLongFunctionOptions = [
+        self::LONG_FUNCTION_ECHO,
+        self::LONG_FUNCTION_PRINT,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        $sample = <<<'EOT'
+<?=1?>
+<?php print '2' . '3'; ?>
+<?php /* comment */ echo '2' . '3'; ?>
+<?php print '2' . '3'; someFunction(); ?>
+
+EOT
+        ;
+
+        return new FixerDefinition(
+            'Replace short-echo `<?=` with long format `<?php echo`/`<?php print` syntax, or vice-versa.',
+            [
+                new CodeSample($sample),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_LONG]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_LONG, self::OPTION_LONG_FUNCTION => self::LONG_FUNCTION_PRINT]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_SHORT]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_SHORT, self::OPTION_SHORT_ALWAYS => true]),
+            ],
+            null
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        if (self::FORMAT_SHORT === $this->configuration[self::OPTION_FORMAT]) {
+            if (null !== $tokens->findSequence([[\T_OPEN_TAG], [\T_ECHO]])) {
+                return true;
+            }
+            if (null !== $tokens->findSequence([[\T_OPEN_TAG], [\T_PRINT]])) {
+                return true;
+            }
+
+            return false;
+        }
+
+        return $tokens->isTokenKindFound(\T_OPEN_TAG_WITH_ECHO);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder(self::OPTION_FORMAT, 'How the fixer should process short/long echo tags'))
+                ->setAllowedValues($this->supportedFormatOptions)
+                ->setDefault(self::FORMAT_LONG)
+                ->getOption(),
+            (new FixerOptionBuilder(self::OPTION_LONG_FUNCTION, 'The function to be used to expand the short echo tags'))
+                ->setAllowedValues($this->supportedLongFunctionOptions)
+                ->setDefault(self::LONG_FUNCTION_ECHO)
+                ->getOption(),
+            (new FixerOptionBuilder(self::OPTION_SHORT_ALWAYS, 'Always render short-echo tags even in case of complex code'))
+                ->setAllowedTypes(['bool'])
+                ->setDefault(false)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(SplFileInfo $file, Tokens $tokens)
+    {
+        if (self::FORMAT_SHORT === $this->configuration[self::OPTION_FORMAT]) {
+            $this->longToShort($file, $tokens);
+        } else {
```

Comment:
There is no point to pass `$file` in above and below lines.

Reply:
Done: f7a7d1a345ad755458c40708a79ec650141f59ef